src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int LIMIT = 100;
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        String randomFigure = TypesOfFigures.values()[random
+                .nextInt(TypesOfFigures.values().length)].name();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Figure figure = new Figure();
+        switch (randomFigure) {
+            case ("CIRCLE"):
+                Circle circle = new Circle();
+                circle.setName("Circle");
+                circle.setRadius(random.nextInt(LIMIT));
+                circle.setColor(colorSupplier.getRandomColor());
+                figure = circle;
+                break;
+            case ("SQUARE"):
+                Square square = new Square();
+                square.setName("Square");
+                square.setSide(random.nextInt(LIMIT));
+                square.setColor(colorSupplier.getRandomColor());
+                figure = square;
+                break;
+            case ("RECTANGLE"):
+                Rectangle rectangle = new Rectangle();
+                rectangle.setName("Rectangle");
+                rectangle.setSideOne(random.nextInt(LIMIT));
+                rectangle.setSideTwo(random.nextInt(LIMIT));
+                rectangle.setColor(colorSupplier.getRandomColor());
+                figure = rectangle;
+                break;
+            case ("RIGHT_TRIANGLE"):
+                RightTriangle rightTriangle = new RightTriangle();
+                rightTriangle.setName("Right Triangle");
+                rightTriangle.setFirstLeg(random.nextInt(LIMIT));
+                rightTriangle.setSecondLeg(random.nextInt(LIMIT));
+                rightTriangle.setColor(colorSupplier.getRandomColor());
+                figure = rightTriangle;
+                break;
```

Comment:
create separate methods for each of the figures that will be returning a random figure, for example, getRandomRightTriangle() and so on